lib/cable_ready/channels.rb

```diff
@@ -8,16 +8,31 @@ def initialize
       @channels = {}
     end
 
-    def [](channel_name)
-      @channels[channel_name] ||= CableReady::Channel.new(channel_name)
+    def [](identifier)
+      @channels[identifier] ||= CableReady::Channel.new(identifier)
     end
 
     def clear
       @channels = {}
     end
 
-    def broadcast
-      @channels.values.map(&:broadcast)
+    def broadcast(identifier = nil)
+      if identifier
+        @channels[identifier].broadcast
+      else
+        @channels.values.select { |channel| channel.identifier.is_a?(String) }.each(&:broadcast)
+      end
+      clear
+    end
+
+    def broadcast_to(model, identifier = nil)
+      if identifier
+        @channels[identifier].broadcast_to model
+      else
+        @channels.values.reject { |channel| channel.identifier.is_a?(String) }.each do |channel|
```

Comment:
I think it should be a `select` here

Reply:
I have the opposite understanding of this. I think it's expecting classes, not strings, which is why you're having problems with your other comment as well.

Reply:
@StephenFiser is correct - this is expecting a valid constant. In your case, `DataChannel` with no quotes.